r-pkg/R/es_search.R

```diff
@@ -100,6 +100,13 @@ es_search <- function(es_host
         )
         log_fatal(msg)
     }
+  
+    # assign 1 core by default, if the number of cores is NA
+    if (is.na(n_cores)){
```

Comment:
could you make this a bit stronger? Now that I think about it, what if it returns `NULL` or some other crazy stuff?

Recommendation:

`if (!assertthat::is.count(n_cores))`

Reply:
oh wow! Really good find!

Yes this might be a bug in `assertthat`. I do know that there have been a bunch of small bugfixes on that project that have not made it to Github yet. Let me see if I can reproduce from the dev version.

For now for this PR, I like your suggestion of the `||` thing

Reply:
@drkarthi ok just checked, definitely a bug!!!

```
git clone https://github.com/hadley/assertthat.git
cd assertthat
R CMD INSTALL .
Rscript -e "assertthat::is.count(NA); assertthat::is.count(NA_real_)"
```

You should [create an issue](https://github.com/hadley/assertthat/issues) and then a PR to fix it! Believe the problem is in the implementation of `is.integerish()` (an internal function used by `assertthat`)

Again, great find!

Reply:
Thank you, James!!! I've just created an issue in asserthat and made a commit with the changes for this PR!

Reply:
https://github.com/hadley/assertthat/issues/54

Reply:
wooo thank you!